OMCompiler/SimulationRuntime/c/util/modelica_string.c

```diff
@@ -304,7 +304,19 @@ extern char* omc__escapedString(const char* str, int nl)
   return res;
 }
 
-int GC_vasprintf(const char **strp, const char *fmt, va_list ap) {
+/**
+ * @brief Garbage collector version of vasprintf.
+ *
+ * Print to newly allocated string.
+ * The duplicated string will be freed by the garbage collector.
+ *
+ * @param strp    Pointer to newly allocated string.
+ * @param fmt     C string containing format strings.
+ * @param ap      Variable argument list for format strings.
+ * @return int    Number of characters written to strp.
+ *                If an error occurs, a negative number is returned.
+ */
+int GC_vasprintf(char **strp, const char *fmt, va_list ap) {
```

Comment:
I think this needs to be `const` because `GC_asprintf` uses it and gets called with `&var`.
```suggestion
int GC_vasprintf(const char **strp, const char *fmt, va_list ap) {
```

Reply:
All of this is not consistent with the non-garbage collected versions of `vasprintf` and `asprintf`:

```C
int asprintf(char **strp, const char *fmt, ...);
int vasprintf(char **strp, const char *fmt, va_list ap);
```

On exit `strp` should contain text from `fmt`, so it is not constant. 
The way to fix it is to change each call of the `GC_vasprintf` (and other) functions. But they are used in a lot of places.